chainer/functions/math/maximum.py

```diff
@@ -18,34 +19,49 @@ def check_type_forward(self, in_types):
         )
 
     def forward_cpu(self, inputs):
+        self.retain_inputs((0, 1))
         x1, x2 = inputs
         y = numpy.maximum(x1, x2)
         return utils.force_array(y),
 
-    def backward_cpu(self, inputs, grads):
-        x1, x2 = inputs
-        gy, = grads
-        gx1 = gy * (x1 >= x2)
-        gx2 = gy * (x1 < x2)
-        return utils.force_array(gx1), utils.force_array(gx2)
-
     def forward_gpu(self, inputs):
+        self.retain_inputs((0, 1))
         x1, x2 = inputs
         return cuda.cupy.maximum(x1, x2),
 
-    def backward_gpu(self, inputs, grads):
-        x1, x2 = inputs
-        gy, = grads
+    def backward(self, indexes, grad_outputs):
+        x1, x2 = self.get_retained_inputs()
+        return MaximumGrad(
+            utils.force_array(x1.data >= x2.data)).apply((grad_outputs[0],))
+
+
+class MaximumGrad(function_node.FunctionNode):
+
+    def __init__(self, cond):
+        self.cond = cond
+
+    def forward_cpu(self, inputs):
+        gy, = inputs
+        gx1 = numpy.where(self.cond, gy, gy.dtype.type(0))
+        gx2 = numpy.where(self.cond, gy.dtype.type(0), gy)
+        return utils.force_array(gx1), utils.force_array(gx2)
+
+    def forward_gpu(self, inputs):
+        gy, = inputs
         gx1 = cuda.elementwise(
-            'T x1, T x2, T gy', 'T gx1',
-            'gx1 = (x1 >= x2) ? gy : (T)0.0',
-            'maximum_bwd1')(x1, x2, gy)
+            'S cond, T gy', 'T gx1',
```

Comment:
How about unify these two kernel to reduce kernel launch cost.

Reply:
Thanks! Fixed as suggested.